tools/osx/xcode_configure.bzl

```diff
@@ -65,6 +64,7 @@ def _xcode_version_output(repository_ctx, name, version, aliases, developer_dir)
             err = xcodebuild_result.stderr,
             out = xcodebuild_result.stdout,
         )
+        fail(error_msg)
```

Comment:
Where does this failure message end up if it's logged in this way? I'm concerned that error messages may be entirely lost which would make xcode_locator failures even harder to debug than they already are.

Reply:
If you ctrl-c out of it, which is the easiest repro case, bazel seems to not print any messages from this. But in that case It also feels like more of an internal bazel failure that the user actually shouldn't need to care about since now it will re-run next time.

It looks like in the analysis failure case, we don't hit this codepath, and this patch doesn't fix that case.

Reply:
If you'd like I can add a print right above this with the same error, as long as we still call `fail()` this should fix that common failure case

Reply:
With the print statement, does a ctrl+C output the print and then fail? If so, lets do that.

Reply:
Yep! it looks like this in that case:

```
Extracting Bazel installation...
Starting local Bazel server and connecting to it...
Analyzing: target //Modules/PassengerApp:Lyft (19 packages loaded, 23 targets configured)
    Fetching @local_config_xcode; fetching
^C
DEBUG: /private/var/tmp/_bazel_ksmiley/c8513c989333be3c89713e16206869b2/external/bazel_tools/tools/osx/xcode_configure.bzl:140:9: Generating xcode-locator-bin failed.  return code 256, stderr: Process terminated by signal 15, stdout:
ERROR: build interrupted
INFO: Elapsed time: 7.614s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (19 packages loaded, 23 targets configured)
    Fetching @local_config_xcode; fetching
```

And on a re-run it's fine. I've updated the PR

Reply:
it does not, the moment you Ctrl-C, it seems to skip all output. Perhaps these could be written into a different file inside the external workspace, something like `errors.log`. As long as there's a `fail()`, Bazel should retry to configure the dependency.

Reply:
What's the status here? Why are Keith and Sergio seeing different behaviors? Is there some flag you need to set to see the debug output?

Reply:
My message above is _after_ I took @c-parsons feedback, without the extra print statement you wouldn't see that line